Snowflake.Data/Core/SnowflakeTomlConnectionBuilder.cs

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Text;
+using Mono.Unix;
+using Mono.Unix.Native;
+using Snowflake.Data.Client;
+using Snowflake.Data.Core.Tools;
+using Snowflake.Data.Log;
+using Tomlyn;
+using Tomlyn.Model;
+
+namespace Snowflake.Data.Core
+{
+    internal class SnowflakeTomlConnectionBuilder
+    {
+        private const string DefaultConnectionName = "default";
+        private const string DefaultSnowflakeFolder = ".snowflake";
+        private const string DefaultTokenPath = "/snowflake/session/token";
+
+        internal const string SnowflakeDefaultConnectionName = "SNOWFLAKE_DEFAULT_CONNECTION_NAME";
+        internal const string SnowflakeHome = "SNOWFLAKE_HOME";
+
+        private readonly SFLogger _logger = SFLoggerFactory.GetLogger<SnowflakeDbConnection>();
+
+        private readonly Dictionary<string, string> _tomlToNetPropertiesMapper = new Dictionary<string, string>(StringComparer.InvariantCultureIgnoreCase)
+        {
+            { "DATABASE", "DB" }
+        };
+
+        private readonly FileOperations _fileOperations;
+        private readonly EnvironmentOperations _environmentOperations;
+
+        internal static readonly SnowflakeTomlConnectionBuilder Instance = new SnowflakeTomlConnectionBuilder();
+
+        internal SnowflakeTomlConnectionBuilder() : this(FileOperations.Instance, EnvironmentOperations.Instance)
+        {
+        }
+
+        internal SnowflakeTomlConnectionBuilder(FileOperations fileOperations, EnvironmentOperations environmentOperations)
+        {
+            _fileOperations = fileOperations;
+            _environmentOperations = environmentOperations;
+        }
+
+        public string GetConnectionStringFromToml(string connectionName = null)
+        {
+            var connectionString = string.Empty;
+            var tomlPath = ResolveConnectionTomlFile();
+            var connectionToml = GetTomlTableFromConfig(tomlPath, connectionName);
+            if (connectionToml != null)
```

Comment:
you could remove the line `var connectionString = string.Empty;`
and instead these lines just write `return connectionToml == null ? string.Empty : GetConnectionStringFromTomlTable(connectionToml)`. It would be more compact. What do you think?

Reply:
Done